metaflow/plugins/aws/step_functions/step_functions.py

```diff
@@ -328,8 +332,62 @@ def _visit(node, workflow, exit_node=None):
                         )
                     )
                     .max_concurrency(self.max_workers)
-                    .output_path("$.[0]")
+                    .output_path("$" if self.use_distributed_map else "$.[0]")
                 )
+
+                # If we are using DistributedMap we need to obtain the Parameters
+                # one of the batch jobs. Distributed Map jobs can be so big that
+                # we can't just pull the value from the OutputPath, instead
+                # we need to write the results to S3 and then get the data from S3.
+                # These additional states are how we pull the data from S3.
+                if self.use_distributed_map:
+                    workflow.add_state_hack(f"{iterator_name}_GetManifest", {
```

Comment:
`add_state_hack` just allowed me to add states that weren't objects but just dictionaries. This isn't something I'd check in, just wanted to get the flow function and get a cursory review before I optimized that functionality.